app/src/main/java/com/giraffe/cineverseapp/di/featureApiModule.kt

```diff
@@ -8,13 +8,32 @@ import com.giraffe.explore.ExploreApi
 import com.giraffe.explore.navigation.ExploreApiImp
 import com.giraffe.home.HomeApi
 import com.giraffe.home.navigation.HomeApiImp
-import org.koin.core.module.dsl.singleOf
-import org.koin.dsl.bind
-import org.koin.dsl.module
-
-val featureApiModule = module {
-    singleOf(::AuthenticationApiImp) bind AuthenticationApi::class
-    singleOf(::DetailsApiImp) bind DetailsApi::class
-    singleOf(::ExploreApiImp) bind ExploreApi::class
-    singleOf(::HomeApiImp) bind HomeApi::class
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object FeatureApiModule {
+
+    @Provides
+    @Singleton
+    fun provideAuthenticationApi(homeApi: HomeApi): AuthenticationApi =
+        AuthenticationApiImp(homeApi)
```

Comment:
why not using  @Binds

```suggestion
    @Binds
    @Singleton
    fun bindsAuthenticationApi(imp: AuthenticationApiImp): AuthenticationApi
```

Reply:
Done, please check it now and let me know if there's anything else.